struct.go

```diff
@@ -115,43 +122,133 @@ type K struct {
 	Data interface{}
 }
 
+func Bool(x bool) *K {
+	return &K{Type: -KB, Attr: NONE, Data: x}
+}
+
+func BoolV(x []bool) *K {
+	return &K{Type: KB, Attr: NONE, Data: x}
+}
+
+func UUID(x uuid.UUID) *K {
+	return &K{Type: -UU, Attr: NONE, Data: x}
+}
+
+func UUIDV(x []uuid.UUID) *K {
+	return &K{Type: UU, Attr: NONE, Data: x}
+}
+
+func Byte(x byte) *K {
+	return &K{Type: -KG, Attr: NONE, Data: x}
+}
+
+func ByteV(x []byte) *K {
+	return &K{Type: KG, Attr: NONE, Data: x}
+}
+
+func Short(x int16) *K {
+	return &K{Type: -KH, Attr: NONE, Data: x}
+}
+
+func ShortV(x []int16) *K {
+	return &K{Type: KH, Attr: NONE, Data: x}
+}
+
 func Int(x int32) *K {
-	return &K{-KI, NONE, x}
+	return &K{Type: -KI, Attr: NONE, Data: x}
+}
+
+func IntV(x []int32) *K {
+	return &K{Type: KI, Attr: NONE, Data: x}
 }
 
 func Long(x int64) *K {
-	return &K{-KJ, NONE, x}
+	return &K{Type: -KJ, Attr: NONE, Data: x}
+}
+
+func LongV(x []int64) *K {
+	return &K{Type: KJ, Attr: NONE, Data: x}
 }
 
 func Real(x float32) *K {
-	return &K{-KE, NONE, x}
+	return &K{Type: -KE, Attr: NONE, Data: x}
+}
+
+func RealV(x []float32) *K {
+	return &K{Type: KE, Attr: NONE, Data: x}
 }
 
 func Float(x float64) *K {
-	return &K{-KF, NONE, x}
+	return &K{Type: -KF, Attr: NONE, Data: x}
 }
 
-func Error(x error) *K {
-	return &K{KERR, NONE, x}
+func FloatV(x []float64) *K {
+	return &K{Type: KF, Attr: NONE, Data: x}
+}
+
+func String(x string) *K {
+	return &K{Type: KC, Attr: NONE, Data: x}
 }
 
 func Symbol(x string) *K {
-	return &K{-KS, NONE, x}
+	return &K{Type: -KS, Attr: NONE, Data: x}
 }
+
 func SymbolV(x []string) *K {
-	return &K{KS, NONE, x}
+	return &K{Type: KS, Attr: NONE, Data: x}
+}
+
+func Timestamp(x time.Time) *K {
+	return &K{Type: -KP, Attr: NONE, Data: x.UTC()}
+}
+
+func TimestampV(x []time.Time) *K {
+	for i, t := range x {
+		x[i] = t.UTC()
+	}
+	return &K{Type: KP, Attr: NONE, Data: x}
+}
+
+func Date(x time.Time) *K {
+	return &K{Type: -KD, Attr: NONE, Data: x.UTC()}
+}
+
+func DateV(x []time.Time) *K {
+	for i, t := range x {
+		x[i] = t.UTC()
+	}
+	return &K{Type: KD, Attr: NONE, Data: x}
+}
+
+func Time(x time.Time) *K {
+	return &K{Type: -KT, Attr: NONE, Data: x.UTC()}
+}
+
+func TimeV(x []time.Time) *K {
+	for i, t := range x {
+		x[i] = t.UTC()
+	}
+	return &K{Type: KT, Attr: NONE, Data: x}
 }
 
-func Atom(t int8, x interface{}) *K {
-	return &K{t, NONE, x}
+func Error(x error) *K {
+	return &K{Type: KERR, Attr: NONE, Data: x}
+}
+
+func Enlist(x ...*K) *K {
```

Comment:
enlist is not the same as NewList in q sense. it is knk(1,x) vs knk(3,x,y,z) if using c api

Reply:
We reverse to revert to NewList